src/pages/Swap/index.tsx

```diff
@@ -698,25 +715,25 @@ export function Swap({
             >
               <ButtonError
                 onClick={() => {
-                  setSwapState({
-                    tradeToConfirm: trade,
-                    swapError: undefined,
-                    showConfirm: true,
-                    txHash: undefined,
-                  })
+                  showPriceImpactWarning
+                    ? setShowPriceImpactModal(true)
+                    : setSwapState({
```

Comment:
nit - this is copy and pasted from above, could you make it a function so people don't have to remember to modify it in two different places